### ISSUE TITLE
Hi (Atom Jumpy author)

### ISSUE THREAD

**Issue by DavidLGoldberg** [wmaurer/vscode-jumpy#18]
Hi,

I'm flattered you enjoyed Jumpy and ported it!  I actually just found out about it today.

The quick check of the code looks really clean, although just quick pass through.  Impressed you did it in so few LOC!

I had loose plans to port it over myself, although I haven't yet played much with VS Code (yet).

I see it's not *completely* full featured yet.  Would you be interested in collaborating with me in future endeavors?  If not, no guarantees I don't make a competing version ;)

No but seriously, I'd love to have a partner in crime!

Currently, right before finding out about this, my plan was to rewrite core Jumpy and Qolor (my other package) in Elm.

Then I think I/we can use "ports" (still only very limited knowledge of Elm) to communicate with the javascript and therefore have access to Atom or VSCode APIs.

Basically, Elm would hold the state machine etc.  I currently have a completely rewritten version (99.5% done but not released) using javascript-state-machine but my goal is to redo that with elm

I noticed that your company lists some functional languages, so any help on that side of things would be extremely helpful too.  No pressure in any of this of course side projects after all...

Would you mind if I email you at your companies email so we can start a thread?

Also, jumpy has a gitter page that would work well too!

**Comment by freMea** [comment]
This extension already has a competitor named MetaGo
https://github.com/metaseed/metaGo

It is currently more featured with selection support.

**Comment by JGH1000** [comment]
Hi @DavidLGoldberg, it seems that the author is no longer maintaining the extension.. Could there be a chance for you to implement jumpy in VSCode? I really like the implementation in Atom, especially that is pick up the theme and the font is really nice and visible.

**Comment by drazik** [comment]
Hello @DavidLGoldberg. It would be awesome if you do an implementation of Jumpy for VSCode. Atom plugin is very simple and useful 👍

**Comment by DavidLGoldberg** [comment]
Should be releasing one in a month or two (estimate hah).

**Comment by brewhue** [comment]
Any update here?

**Comment by DavidLGoldberg** [comment]
Ah, It'll be a little while sorry.

I have made some strives towards it..actually shortly I should be releasing
(with Atom Jumpy) this core lib written in elm....that can be used across
both IDEs ...as a core state machine....and the rest of the arch I believe
is VERY portable..

If you are looking to help contrib, wait a bit until I fix some spec test
issues in the atom version and then I'm going to launch it (that's going to
take priority)

I'm then going to switch gears to VS Code version for sure if you want to
pair up a bit...

I don't have the branch pushed at the moment, but I've tested (manually all
of the functionality)... I just have to make some asynch stuff work in the
tests.

**Comment by andys8** [comment]
@DavidLGoldberg Sounds promising. Escpecially written in Elm, because I'm a big fan of the Elm language. Will definitely have a look as soon as you've released it.

**Comment by opsb** [comment]
@DavidLGoldberg would love to see this if you got anywhere with it (also a big Elm fan)

**Comment by DavidLGoldberg** [comment]
@opsb @andys8 @ajitid @drazik 
If you happen to still be using VS Code and interested:
https://marketplace.visualstudio.com/items?itemName=DavidLGoldberg.jumpy2

Sorry for the delays, but I do plan on supporting this going forward (I never really stopped supporting Atom's its just pretty stable), and this new one is *almost* parity with Atom.

Important features are implemented, sans tab and tree view jumps, which look difficult with VS Code.

Please rate it 5 stars and star the GH repo if you like it :)

Oh yeah there's a Youtube video you can subscribe/like too hah!

Please any feedback is really, really appreciated! (transfer to new repo's issues / discussions feature of course!!)
